src/ring_theory/ideal/basic.lean

```diff
@@ -402,7 +391,7 @@ theorem span_singleton_prime {p : α} (hp : p ≠ 0) :
   is_prime (span ({p} : set α)) ↔ prime p :=
 by simp [is_prime_iff, prime, span_singleton_eq_top, hp, mem_span_singleton]
 
-theorem is_maximal.is_prime {I : ideal α} (H : I.is_maximal) : I.is_prime :=
+theorem _root_.submodule.is_maximal.is_prime {I : ideal α} (H : I.is_maximal) : I.is_prime :=
```

Comment:
Why not make a `@[reducible] def ideal.is_maximal := submodule.is_maximal` so we can keep the original names of lemmas? (I think dot notation for `submodule.is_maximal` would still work for `ideal.is_maximal` per reducibility.)

Reply:
This crossed my mind too. I think we still end up in a bit of a mess if we end up with `submodule.is_maximal I` and need something from the `ideal.is_maximal` namespace.

Reply:
Well, `ideal` is already reducibly defeq to `submodule`...